pkg/kubelet/pod/pod_manager.go

```diff
@@ -154,6 +159,11 @@ func (pm *basicManager) UpdatePod(pod *v1.Pod) {
 	pm.lock.Lock()
 	defer pm.lock.Unlock()
 	pm.updatePodsInternal(pod)
+	if pm.checkpointManager != nil {
+		if err := pm.checkpointManager.WritePod(pod); err != nil {
+			glog.Errorf("Error writing checkpoint for pod: %v", pod.GetName())
```

Comment:
What's the expected behavior if this happens? kubelet simply won't restore the pod at startup?

Reply:
Yes, at some point intervention will be required, and given this is for bootstrapping this is ok.